### ISSUE TITLE
Catch unhandled promises

### ISSUE THREAD

**Issue by IPWright** [mperdeck/jsnlog#131]
Out of the box JSNLog catches errors that bubble up to `window.onerror` which is great, however it doesn't catch any errors within promises. 

I can see an argument for not catching errors in promises that handle rejection, but any that don't really should be caught by JSNLog. I've mocked together some code for what I'm doing, something like this might fix this (note not all browsers support `onunhandledrejection`:

```
if (typeof window !== 'undefined' && !window.onunhandledrejection) {
    window.onunhandledrejection = function(event) {//= function (errorMsg, url, lineNumber, column, errorObj) {
        
        // Send object with all data to server side log, using severity fatal, 
        // from logger "onerrorLogger"
        JL("onerrorLogger").fatalException({
            "msg": "unhandledrejection",
            "errorMsg": event.reason.message
        }, event.reason);
        // Tell browser to run its own error handler as well   
        return false;
    };
}
```

**Comment by mperdeck** [comment]
Thanks for the suggestion. I'll probably have time coming weekend to implement this.

I'll have to test this with a promise that throws an unhandled exception. Do you happen to have a snippet that does this?

**Comment by mperdeck** [comment]
I've added this to JSNLog, and released it in version 2.22.0.
